### DISCUSSION TITLE
How much does spdlog relies on STL?

### DISCUSSION THREAD

**Discussion by lgacnik** [gabime/spdlog#2830] [Q&A]
I'm looking for a lightweight, fast, and thread-safe logger which for an embedded device. I see that spdlog uses Standard Template Library which is something I want to avoid for now. However if spdlog doesn't heavily rely on STL I would consider trying it out. Note that I'm not knowledgable about STL, I'm just wary of it since my hardware specs are limited. So, generally I would ask whether spdlog logger heavily relies on STL or/and is spdlog applicable for use on embedded devices as well?

**Comment by tt4g** [comment]
No, STL strongly dependent.
STL is such a fundamental feature of the C++ language that not depending on STL can be described as a plain C language.
